platform/web/js/libs/library_godot_os.js

```diff
@@ -346,6 +346,16 @@ const GodotOS = {
 		return concurrency < 2 ? concurrency : 2;
 	},
 
+	godot_js_os_pthread_pool_size_get__proxy: 'sync',
+	godot_js_os_pthread_pool_size_get__sig: 'i',
+	godot_js_os_pthread_pool_size_get: function () {
+		if (typeof PThread !== 'undefined' && PThread.runningWorkers && PThread.unusedWorkers) {
+			const total = PThread.runningWorkers.length + PThread.unusedWorkers.length;
+			return Math.max(1, total - 1);
+		}
+		return 1;
+	},
```

Comment:
```suggestion
	godot_js_os_pthread_pool_size_get: function () {
		if (typeof PThread === 'undefined') {
		    // Threads aren't supported, so default to `1`.
			return 1;
		}
		const total = PThread.runningWorkers.length + PThread.unusedWorkers.length;
		return Math.max(1, total - 1);
	},
```

- I think we should always prefer early exiting.
- `if (PThread == null)` could be valid too, but I'm not _that_ nitpicky.

It seems that you can trust that `Pthread.{runningWorkers,unusedWorkers}` exist. I didn't see anything [in the emscripten source code](https://github.com/emscripten-core/emscripten/) that suggests that these could be anything else than arrays.

Reply:
I agree, I was being a bit more defensive than usual since wasn't sure in which emscripten version it was added.

Reply:
Oh, didn't think about that. Mmmm. I'll investigate.

Reply:
https://github.com/emscripten-core/emscripten/blame/62d4b1cc0b98d2a2ccdd59e0fc24d76703e1d866/src/lib/libpthread.js#L79-L81

I think we're OK! :)